src/modules/CRUD/services/CRUDProvider.js

```diff
@@ -29,105 +33,174 @@ export class CRUDProvider extends React.Component {
   setMap = map => !this.isUnmount && this.setState({ map });
 
   getMapConfig = async () => {
+    const result = {};
     try {
-      const { results: mapConfig = {} } = await fetchMapConfig();
-      this.setState({ mapConfig });
+      const mapConfig = await fetchMapConfig();
+      result.mapConfig = mapConfig;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, code: e.message },
-      }));
+      result.error = e;
     }
+
+    const { mapConfig = {}, error } = result;
+    this.setState(({ errors }) => ({
+      mapConfig,
+      errors: {
+        ...errors,
+        mapConfig: error,
+      },
+    }));
+
+    return mapConfig;
   };
 
   getSettings = async () => {
+    const result = {};
     try {
       const settings = await fetchSettings();
-      this.setState({ settings });
+      result.settings = settings;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, code: e.message },
-      }));
+      result.error = e;
+    }
+
+    const { settings = {}, error } = result;
+
+    this.setState(({ errors }) => ({
+      settings,
+      errors: {
+        ...errors,
+        settings: error,
+      },
+    }));
+
+    return settings;
+  }
+
+  getFormattedError = props => {
+    const {
+      store,
+      error,
+      ids,
+    } = props;
+    const { errors: { [store]: errorStore } } = this.state;
+
+    const propsFromIds = Object.keys(ids);
+
+    // Remove error if it was before in the store
+    if (!(error instanceof Error)) {
+      return errorStore.filter(errorItem => (
+        propsFromIds.every(prop => errorItem[prop] !== ids[prop])
+      ));
     }
+
+    // Modify error if it was before in the store
+    const isErrorItemAlreadyExisting = errorStore.some(errorItem => (
+      propsFromIds.every(prop => errorItem[prop] === ids[prop])
+    ));
+
+    if (isErrorItemAlreadyExisting) {
+      return errorStore.map(errorItem => (
+        propsFromIds.every(prop => errorItem[prop] === ids[prop])
+          ? { error, ...ids }
+          : errorItem
+      ));
+    }
+
+    // Else add error in the store
+    return [...errorStore, { error, ...ids }];
+  }
+
+  onlyExistingFeatures = (featuresList, feature) => {
```

Comment:
`onlyExistingFeatures`is a _pure function_, it should be defined as a component method.

Reply:
I rename this function with `updateOrSaveFeatureInFeaturesList` and now it's exported from `features.js`